### PR TITLE
Update helm charts to be consistent with mesheryctl

### LINKED ISSUE #4202: Broker and Meshsync pods, along with kube-rbac-proxy container are missing while installing with helm chart
#### Current Behavior

The `Broker` and `Meshsync` pods are missing after meshery is deployed using the [helm chart](https://github.com/meshery/meshery/tree/master/install/kubernetes/helm/meshery)

The `kube-rbac-proxy` container is also missing in the `meshery-operator` pod compared to the installation result that `mesheryctl` gives 
#### Expected Behavior

We should provide a way to manually create CR instances for `Broker` and `Meshsync` after the CRDs are installed

We should also add the `kube-rbac-proxy` container to the `meshery-operator` pod so that the meshery installation process with helm chart is consistent with `mesheryctl`

#### Screenshots/Logs

With helm chart installation:
```
helm install test install/kubernetes/helm/meshery -n meshery --kubeconfig=$KUBECONFIG
NAME: test
LAST DEPLOYED: Wed Sep 15 15:07:05 2021
NAMESPACE: meshery
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Get the application URL by running these commands:
     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
           You can watch the status of by running 'kubectl get --namespace meshery svc -w test-meshery'
  export SERVICE_IP=$(kubectl get svc --namespace meshery test-meshery --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}")
  echo http://$SERVICE_IP:9081/provider
2. All the Notes.txt of Meshery, Mesheryistio and Mesherylinkerd need to add the content here. This can be achieved by sub PR.

kubectl get pods -n meshery --kubeconfig=$KUBECONFIG
NAME                                    READY   STATUS    RESTARTS   AGE
meshery-677df7b644-kjg68                1/1     Running   0          76s
meshery-consul-5769bf7c9c-vqk6j         1/1     Running   0          77s
meshery-cpx-679cb4c5bb-5z8vc            1/1     Running   0          77s
meshery-istio-55bc6b4679-bhh84          1/1     Running   0          77s
meshery-kuma-64476f65fb-vzf98           1/1     Running   0          76s
meshery-linkerd-bb74b8dd9-bs6s9         1/1     Running   0          76s
meshery-nsm-846c648c97-xx8vw            1/1     Running   0          76s
meshery-operator-8899b86cf-cjsb4        1/1     Running   0          76s
meshery-osm-97995cd6f-pjfdd             1/1     Running   0          76s
meshery-traefik-mesh-65bc8888bc-nk5mn   1/1     Running   0          76s

kubectl get brokers.meshery.layer5.io -n meshery
No resources found in meshery namespace.

kubectl get meshsyncs.meshery.layer5.io -n meshery
No resources found in meshery namespace.
```

With `mesheryctl` installation:
```
kubectl get pods -n meshery
NAME                                    READY   STATUS    RESTARTS   AGE
meshery-58f5449cc-tjrmq                 1/1     Running   0          68s
meshery-broker-0                        2/2     Running   0          27s
meshery-consul-6cb6f89f66-wbjk2         1/1     Running   0          62s
meshery-cpx-8497c64cfc-2gh7v            1/1     Running   0          57s
meshery-istio-db9b99584-mhnrz           1/1     Running   0          66s
meshery-kuma-9b5ddbb77-dmnmz            1/1     Running   0          59s
meshery-linkerd-94c765956-swpjn         1/1     Running   0          64s
meshery-meshsync-5bcbc874f8-rzgsg       1/1     Running   2          25s
meshery-nginx-sm-5546bc9b5d-2qcxp       1/1     Running   0          52s
meshery-nsm-c7fd64dc4-nqdrd             1/1     Running   0          61s
meshery-operator-bddf9fd66-8sr68        2/2     Running   0          34s
meshery-osm-5ffd986b64-z674l            1/1     Running   0          56s
meshery-traefik-mesh-5d6695b5fd-zs5rl   1/1     Running   0          54s

kubectl get brokers.meshery.layer5.io -n meshery
NAME             AGE
meshery-broker   8m11s

kubectl get meshsyncs.meshery.layer5.io -n meshery
NAME               AGE
meshery-meshsync   8m46s
```
#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:**  Kubernetes
- **Meshery Server Version:** stable-v
- **K8s version** any
- **Helm version** any

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by panyuenlau**
Signed-off-by: Darren Lau panyuenlau@gmail.com

**Description**

This PR fixes #4202 

**Notes for Reviewers**
First install meshery-oeprator along with all the adapters and the CRDs for Broker and Meshsync:
```
helm install meshery install/kubernetes/helm/meshery -n meshery
NAME: meshery
LAST DEPLOYED: Sun Sep 19 13:53:04 2021
NAMESPACE: meshery
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Get the application URL by running these commands:
     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
           You can watch the status of by running 'kubectl get --namespace meshery svc -w meshery'
  export SERVICE_IP=$(kubectl get svc --namespace meshery meshery --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}")
  echo http://$SERVICE_IP:9081/provider
2. All the Notes.txt of Meshery, Mesheryistio and Mesherylinkerd need to add the content here. This can be achieved by sub PR.
```

Then install the CR instances for Broker and Meshsync:
```
helm install broker install/kubernetes/helm/meshery-cr/meshery-broker/ -n meshery
NAME: broker
LAST DEPLOYED: Sun Sep 19 21:25:52 2021
NAMESPACE: meshery
STATUS: deployed
REVISION: 1
TEST SUITE: None

helm install meshsync install/kubernetes/helm/meshery-cr/meshery-meshsync/ -n meshery 
NAME: meshsync
LAST DEPLOYED: Sun Sep 19 21:26:08 2021
NAMESPACE: meshery
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

**Results**
The resources that are deployed using the helm charts match with the `mesheryctl`:
```
kubectl get pods -n meshery
NAME                                    READY   STATUS    RESTARTS   AGE
meshery-5cc4489f77-kvkzf                1/1     Running   0          7h38m
meshery-broker-0                        2/2     Running   0          6m34s
meshery-consul-77d459ccf6-g8lkg         1/1     Running   0          7h38m
meshery-cpx-6b54ff5ffc-46bfj            1/1     Running   0          7h38m
meshery-istio-6c56dd44fb-6hz9q          1/1     Running   0          7h38m
meshery-kuma-7f4dbc98cc-2dnf5           1/1     Running   0          7h38m
meshery-linkerd-5456b548cd-v8xsd        1/1     Running   0          7h38m
meshery-meshsync-86d4959b54-fc2qb       1/1     Running   2          6m22s
meshery-nsm-5bfd567998-7l4lc            1/1     Running   0          7h38m
meshery-operator-7486c6bddc-dcjcb       2/2     Running   0          7h38m
meshery-osm-55d95c7b8b-jfdrg            1/1     Running   0          7h38m
meshery-traefik-mesh-64bb9c58b6-65wd7   1/1     Running   0          7h38m
```
After the Broker and Meshsync pods are created, the meshery-oeprator is able to reconcile them:

> _Note_:
> It seems the `mesheryctl` uses the yamls in [meshery-operator/config](https://github.com/meshery/meshery-operator/tree/master/config) and [meshery/install/deployment_yamls](https://github.com/meshery/meshery/tree/master/install/deployment_yamls/k8s) to do the installation (source code reference: https://github.com/meshery/meshery/blob/master/mesheryctl/pkg/utils/helpers.go#L87-L104), this is probably not the best practice because it would cause inconsistencies and confusions because. 
>
> It'd be better to refactor the `mesheryctl` implementation to use the helm charts to do the installation so that using the `mesheryctl` is just an automated way of using `helm` charts to do the installations.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: install/kubernetes/helm/meshery/charts/meshery-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -25,12 +25,12 @@ spec:
...
+        - name: {{ .Values.mesheryOperator.name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
           # TODO: Need to confirm the meshery-operator image version
-          image: "{{ .Values.image.repository }}"
-          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          image: "{{ .Values.mesheryOperator.image.repository }}"
+          imagePullPolicy: {{ .Values.mesheryOperator.image.pullPolicy }}
           ports:
             - name: http
```
Comment (panyuenlau) [inline install/kubernetes/helm/meshery/charts/meshery-operator/templates/deployment.yaml:35]: I noticed that the `ports` that we have for the `manager` container in this helm chart doesn't match the manifest that `mesheryctl` uses: https://github.com/meshery/meshery-operator/blob/master/config/manifests/default.yaml#L425-L430, probably need to update this to match the YAML that `meshseryctl` uses as well

**Comment by leecalcote** [comment]
This is great. Consider these two objectives and how we might simultaneously achieve them:

1) As few commands as possible to install. Ideally, one command. Ideally, we have a version of the chart that includes all containerized Meshery components: Meshery Server, Adapters, and Operator (+controllers).

2) The ability to independently deploy Meshery Operator is desired. Once Meshery has been deployed to an initial cluster and as users connect subsequent Kubernetes clusters (bring other clusters under Meshery's management), A) Meshery Server will need to deploy the Operator independently and B) ideally, we allow users to deploy Meshery Operator independently.

**Review by Aisuko — COMMENTED** [review]
Please kindly consider the suggestion from the Slack channel https://layer5io.slack.com/archives/CFGG6U10E/p1632436124030600?thread_ts=1632324551.019600&cid=CFGG6U10E

**Comment by panyuenlau** [comment]
Now it only requires one `helm install` command to install all the components to the cluster, for example:
```
helm install meshery install/kubernetes/helm/meshery -n meshery

kubectl get pods -n meshery 
NAME                                    READY   STATUS    RESTARTS   AGE
meshery-5cc4489f77-9fqdb                1/1     Running   0          6m2s
meshery-broker-0                        2/2     Running   0          13m
meshery-consul-77d459ccf6-28mnw         1/1     Running   0          6m2s
meshery-cpx-6b54ff5ffc-qrhjt            1/1     Running   0          6m2s
meshery-istio-6c56dd44fb-9hxjr          1/1     Running   0          6m2s
meshery-kuma-7f4dbc98cc-62jl9           1/1     Running   0          6m2s
meshery-linkerd-5456b548cd-qnfc2        1/1     Running   0          6m2s
meshery-meshsync-5969865968-5h2cr       1/1     Running   0          5m59s
meshery-nsm-5bfd567998-cdl72            1/1     Running   0          6m2s
meshery-operator-586c6f895d-t79vw       2/2     Running   0          6m2s
meshery-osm-55d95c7b8b-6567q            1/1     Running   0          6m2s
meshery-traefik-mesh-64bb9c58b6-7qqmt   1/1     Running   0          6m2s
```

File: install/kubernetes/helm/meshery/Chart.yaml [hunk 2]
```diff
@@ -13,14 +13,17 @@ keywords:
   - meshery-nsm
   - meshery-traefik-mesh
   - meshery-cpx
+  - meshery-nginx-sm
   - meshery-operator
-source: 
+  - meshery-broker
+  - meshery-meshsync
+source:
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/Chart.yaml:20]: 3 new charts. Nice.

File: install/kubernetes/helm/meshery/Chart.yaml [hunk 3]
```diff
@@ -77,3 +89,5 @@ maintainers:
     email: urakiny@gmail.com
   - name: leecalcote
     email: lee.calcote@layer5.io
+  - name: darrenlau
+    email: panyuenlau@gmail.com
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/Chart.yaml:93]: Excellent.🥇

File: install/kubernetes/helm/meshery/charts/meshery-meshsync/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v2
+name: meshery-meshsync
+description: Meshery Meshsync CR Chart
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/charts/meshery-meshsync/Chart.yaml:3]:
```suggestion
description: Meshery MeshSync CR Chart
```

File: install/kubernetes/helm/meshery/charts/meshery-nginx-sm/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+apiVersion: v2
+name: meshery-nginx-sm
+description: Meshery Adapter for Nginx Service Mesh.
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/charts/meshery-nginx-sm/Chart.yaml:3]:
```suggestion
description: Meshery Adapter for NGINX Service Mesh.
```

File: install/kubernetes/helm/meshery/charts/meshery-nginx-sm/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# meshery-nginx-sm
+
+![Version: 1.0.1](https://img.shields.io/badge/Version-1.0.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: stable-latest](https://img.shields.io/badge/AppVersion-stable--latest-informational?style=flat-square)
+
+Meshery Adapter for Nginx Service Mesh.
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/charts/meshery-nginx-sm/README.md:5]:
```suggestion
Meshery Adapter for NGINX Service Mesh.
```

File: install/kubernetes/helm/meshery/charts/meshery-operator/values.yaml [hunk 7]
```diff
@@ -4,9 +4,27 @@
...
 replicaCount: 1
 
-image:
-  repository: layer5/meshery-operator:stable-latest
-  pullPolicy: Always
+mesheryOperator:
+  name: manager
+  image:
+    repository: layer5/meshery-operator:stable-latest
+    pullPolicy: Always
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery/charts/meshery-operator/values.yaml:11]: Good stuff here.

**Review by leecalcote — APPROVED** [review]
@jbyers a chart for NGINX SM...

**Comment by leecalcote** [comment]
Way to be on this, @panyuenlau. 💪

**Pull request merged**